internal/services/iam/ssh_key.go

```diff
@@ -55,6 +55,14 @@ func ResourceSSKKey() *schema.Resource {
 					areEqual := bytes.Equal(marshalledOldValue, marshalledNewValue)
 					return areEqual
 				},
+				StateFunc: func(v interface{}) string {
```

Comment:
Could you add a bit of documentation explaining that the last part of the public key will be trimmed.